pytest_django/plugin.py

```diff
@@ -22,10 +22,71 @@
 
 SETTINGS_MODULE_ENV = 'DJANGO_SETTINGS_MODULE'
 CONFIGURATION_ENV = 'DJANGO_CONFIGURATION'
-
+DJANGO_ASSERTS = {
+    (1, 3): ('assertContains', 'assertFormError', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRedirects',
+             'assertTemplateNotUsed', 'assertTemplateUsed'),
+    (1, 4): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertHTMLEqual', 'assertHTMLNotEqual', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRaisesMessage',
+             'assertRedirects', 'assertTemplateNotUsed', 'assertTemplateUsed'),
+    (1, 5): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertHTMLEqual', 'assertHTMLNotEqual', 'assertInHTML',
+             'assertJSONEqual', 'assertNotContains', 'assertNumQueries',
+             'assertQuerysetEqual', 'assertRaisesMessage', 'assertRedirects',
+             'assertTemplateNotUsed', 'assertTemplateUsed', 'assertXMLEqual',
+             'assertXMLNotEqual'),
+    (1, 6): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRaisesMessage',
+             'assertRedirects', 'assertTemplateNotUsed', 'assertTemplateUsed',
+             'assertXMLEqual', 'assertXMLNotEqual'),
+    (1, 7): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertNotContains',
+             'assertNumQueries', 'assertQuerysetEqual', 'assertRaisesMessage',
+             'assertRedirects', 'assertTemplateNotUsed', 'assertTemplateUsed',
+             'assertXMLEqual', 'assertXMLNotEqual'),
+    (1, 8): ('assertContains', 'assertFieldOutput', 'assertFormError',
+             'assertFormsetError', 'assertHTMLEqual', 'assertHTMLNotEqual',
+             'assertInHTML', 'assertJSONEqual', 'assertJSONNotEqual',
+             'assertNotContains', 'assertNumQueries', 'assertQuerysetEqual',
+             'assertRaisesMessage', 'assertRedirects', 'assertTemplateNotUsed',
+             'assertTemplateUsed', 'assertXMLEqual', 'assertXMLNotEqual')
+}
```

Comment:
I believe it should be possible to collect these method directly from django's SimpleTestCase, TestCase and TransactionalTestCase classes rather than hardcoding.